src/util/parse-arguments.util.ts

```diff
@@ -0,0 +1,28 @@
+export function parseArgs(argv: string[]): Record<string, string[]> {
+    const parsed: Record<string, string[]> = {};
+
+    const args: string[] = argv.slice(2);
+    const indicesOfFlags: number[] = [];
+
+    for (let i = 0; i < args.length; i++) {
```

Comment:
I think this could be done in a single reduce:
```
// Do we need to allow multiple values per key? Or do we just take the last one? `Record<string, string>` might suffice
return args.reduce((prev: Record<string, string[]>, curr: string): Record<string, string[]> {
    const parsedKey: string = curr.replaceOrWhatever(...); // Might be worth its own function, for testability. If we have multiple functions in this file, I suggest to create a `ParseArgsUtil` that exposes `parseArgs` as public static member)
    const parsedValue: string = curr.replaceOrWhatever(...); // same as above, separate method
    
    return {
        ...prev,
        [parsedKey]: [...(prev[parsedKey] ?? []), parsedValue]
    };
}, {});
```

**EDIT**: Ah, nevermind, I didn't quite get how it works - a single arg-entry isn't already a key-value-pair.
Oof... what kind of API do we want to support exactly? :D 

In that case, I think going for a more "strict" approach might also work:
```
type allowedKeysType = keyof cliSettings;
let currentKey: allowedKeysType | null = null;
return args.reduce((prev: cliSettings, curr: string): cliSettings => {
    if(curr === '--checkDirs) {
        currentKey = 'checkDirectories'; // or whatever it was
        return prev;
    }
    // Also if-checks for all the other key-possibilities, could be done more elegant but doesn't matter right now
    if(!currentKey) {
         // this is an error here I guess
         return prev;
    }
    return {
        ...prev,
        [currentKey]: curr // or concatenate curr or whatever
    };
}, emptyCliSettings);
```